ui/packages/consul-ui/app/services/repository/dc.js

```diff
@@ -1,47 +1,29 @@
-import { inject as service } from '@ember/service';
-import RepositoryService from 'consul-ui/services/repository';
-import { get } from '@ember/object';
 import Error from '@ember/error';
+import RepositoryService from 'consul-ui/services/repository';
 import dataSource from 'consul-ui/decorators/data-source';
 
 const modelName = 'dc';
 export default class DcService extends RepositoryService {
-  @service('settings') settings;
-  @service('env') env;
-
   getModelName() {
     return modelName;
   }
 
-  @dataSource('/:ns/:dc/datacenters')
-  async findAll(params, configuration = {}) {
-    return this.store.query(this.getModelName(), {});
+  @dataSource('/:partition/:ns/:dc/datacenters')
+  async findAll() {
+    return super.findAll(...arguments);
   }
 
-  async findBySlug(name) {
+  @dataSource('/:partition/:ns/:dc/datacenter/:name')
+  async findBySlug(params) {
     const items = this.store.peekAll('dc');
-    if (name != null) {
-      const item = await items.findBy('Name', name);
-      if (typeof item !== 'undefined') {
-        return item;
-      }
+    const item = items.findBy('Name', params.name);
+    if (typeof item === 'undefined') {
+      // FIXME: HTTPError
+      const e = new Error('Page not found');
+      e.status = '404';
+      // FIXME: EDError
+      throw { errors: [e] };
```

Comment:
This should use proper errors instead of an object. I want to stop using the custom ember-data shaped errors as we don't need them. So ideally we'd just go back to using a standard shaped error.